### PR TITLE
Cardano multisig review fixes

### PR DESCRIPTION
**Pull request by davidmisiak**

### CODE REVIEW

File: core/src/apps/cardano/sign_tx.py [hunk 1]
```diff
@@ -787,6 +774,7 @@ def _should_show_output(
     signing_mode: CardanoTxSigningMode,
 ) -> bool:
     if signing_mode == CardanoTxSigningMode.POOL_REGISTRATION_AS_OWNER:
+        # there are no inputs belonging to the user and no spending witnesses
```
Comment (gabrielKerekes) [inline core/src/apps/cardano/sign_tx.py:777]:
```suggestion
        # In a pool registration transaction, there are no inputs belonging to the user
        # and no spending witnesses. It is thus safe to not show the outputs.
```

File: core/src/apps/cardano/sign_tx.py [hunk 2]
```diff
@@ -812,25 +800,22 @@ async def _show_output(
...
+        payment_credential_policy = get_payment_credential_policy(address_parameters)
+        stake_credential_policy = get_stake_credential_policy(
             keychain, address_parameters
         )
 
         await show_credential(
             ctx,
             CredentialParams(CredentialParams.TYPE_PAYMENT, address_parameters),
             payment_credential_policy,
+            True,
```
Comment (gabrielKerekes) [inline core/src/apps/cardano/sign_tx.py:812]: I'd rather include the name of the parameter in this case. The same applies also below. 
```suggestion
            is_change_output=True,
```

**Review by gabrielKerekes — COMMENTED** [review]
Just some small suggestions. I'll do a second review when the policies are updated.

File: core/src/apps/cardano/helpers/address_credential_policy.py [hunk 3]
```diff
@@ -9,90 +8,80 @@
...
+    # fmt: on
+
+    @classmethod
+    def maybe_unusual(cls, is_unusual: bool, policy: int) -> int:
+        """Helper function that optionally adds FLAG_WARN and FLAG_UNUSUAL_PATH flags."""
+        return policy | cls.WARN_UNUSUAL_PATH if is_unusual else policy
+
+
 def get_address_policy(
     keychain: Keychain, address_parameters: CardanoAddressParametersType
```
Comment (gabrielKerekes) [inline core/src/apps/cardano/helpers/address_credential_policy.py:43]: The `keychain` parameter is now unused. Please also check callers of this function if they need the parameter.

Comment (davidmisiak) [reply core/src/apps/cardano/helpers/address_credential_policy.py:43]: Hmm, I thought `make pystyle_check` would catch unused arguments. Thanks for noticing, fixed.

File: core/src/apps/cardano/helpers/address_credential_policy.py [hunk 4]
```diff
@@ -9,90 +8,80 @@
...
-    CREDENTIAL_POLICY_WARN_NO_STAKING,
-    CREDENTIAL_POLICY_FAIL_OR_WARN_UNUSUAL,
-    CREDENTIAL_POLICY_WARN_MISMATCH,
-)
-
 ADDRESS_POLICY_SHOW_SPLIT = 0
 ADDRESS_POLICY_SHOW_SIMPLE = 1
 
 
+class CredentialPolicy:
```
[the remark this answers is not in the document]
Comment (davidmisiak) [reply core/src/apps/cardano/helpers/address_credential_policy.py:15]: Should be fixed.

File: core/src/apps/cardano/helpers/address_credential_policy.py [hunk 5]
```diff
@@ -9,90 +8,80 @@
...
     ):
-        return CREDENTIAL_POLICY_WARN_SCRIPT
+        return C.WARN_GENERAL
 
     elif address_parameters.address_type in (
         CardanoAddressType.REWARD,
         CardanoAddressType.REWARD_SCRIPT,
     ):
-        return CREDENTIAL_POLICY_WARN_REWARD_ADDRESS
+        return C.WARN_REWARD
```
[the remark this answers is not in the document]
Comment (davidmisiak) [reply core/src/apps/cardano/helpers/address_credential_policy.py:84]: Fixed.

**Review by gabrielKerekes — CHANGES_REQUESTED** [review]

File: core/tests/test_apps.cardano.address_credential_policy.py [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+    CredentialPolicy,
+    get_payment_credential_policy,
+    get_stake_credential_policy,
+)
+from apps.common.paths import HARDENED
+from trezor.enums import CardanoAddressType
+from trezor.messages import CardanoAddressParametersType, CardanoBlockchainPointerType
+
+
+def _harden(path: list[int]) -> list[int]:
```
Comment (gabrielKerekes) [inline core/tests/test_apps.cardano.address_credential_policy.py:13]: I'd rather not introduce another method of "hardening". I'd prefer to use what's used in `test_apps.cardano.certificate.py`. This way it's not really clear what's being hardened on first sight - although yes, it's fairly obvious if you know what should be being hardened. But let's not assume everyone knows.

Comment (davidmisiak) [reply core/tests/test_apps.cardano.address_credential_policy.py:13]: Ok, fixed. I wanted to make it a bit more readable, but it doesn't really matter.

**Pull request merged**